tuf/data/keys.go

```diff
@@ -317,113 +327,226 @@ func NewED25519PublicKey(public []byte) *ED25519PublicKey {
 }
 
 // Private key types
+type privateKey struct {
+	private []byte
+}
+
+type signer struct {
+	Signer crypto.Signer
```

Comment:
Non-blocking - not sure if it might be kind of confusing to have multiple ways of getting this - since it's already exposed on every PrivateKey with CryptoSigner(), maybe it'd be simpler to not expose this?